source/common/upstream/cluster_manager_impl.h

```diff
@@ -361,14 +367,38 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
   // This map is ordered so that config dumping is consistent.
   typedef std::map<std::string, ClusterDataPtr> ClusterMap;
 
+  struct PendingUpdates {
+    Event::TimerPtr timer_;
+    bool timer_enabled_{};
+    MonotonicTime last_updated_;
```

Comment:
I'm not sure if `MonotonicTime` is a typedef to a primitive type or not. Should this be 0 initialized or does it not matter? Would the code actually be more clear if this was an absl::optional to indicate on initial creation there is no last update?

Reply:
`MonotonicTime` is:

```
typedef std::chrono::time_point<std::chrono::steady_clock> MonotonicTime;
```

and, by design, we rely on the default constructor which creates a `time_point` with the clock's epoch (which probably is the computer's boot time, though this is not specified). I'll drop a comment to explain this.